### DISCUSSION TITLE
Handler for Unimplemented Error

### DISCUSSION THREAD

**Discussion by peterkuykendall** [swaggest/rest#104] [Q&A] [answered]
Could you also please share an example of a handler that returns a simple error only.

The particular use case (in the product management sense) that I'm trying to cover is a handler that always returns an Unimplemented status and message that I can put in place during development to cover each of the end points that I still need to build out.  Once I write the actual endpoint handler, I'll just re-point to the new handler.

**Comment by vearutop** [comment, marked as answer]
Please try such `usecase.Interactor`.

```go
func unimplemented() usecase.Interactor {
	u := usecase.NewInteractor(func(ctx context.Context, in struct{}, out *struct{}) error {
		return status.Unimplemented

		// Or add some message to the error.
		//return status.Wrap(errors.New("please come later"), status.Unimplemented)
	})

	u.SetExpectedErrors(status.Unimplemented)

	return u
}
```

**Reply by peterkuykendall** [reply]
wonderful -- thanks! I wasn't using the SetExpectedErrors.
